### ISSUE TITLE
Crest 2.11 does not build with gfortran 11

### ISSUE THREAD

**Issue by susilehtola** [grimme-lab/crest#53]
Using CMake 3.19.7 and gfortran 11.1.1 on Fedora 34, the build fails in
```
[ 12%] Building Fortran object CMakeFiles/crest-exe.dir/src/zdata.f90.o
/home/susi/rpmbuild/BUILD/crest-2.11/src/zdata.f90:88:33:

   88 |          procedure deallocate => deallocate_zring  !deallocate (if allocated)
      |                                 1
Error: ‘::’ needed in PROCEDURE binding with explicit target at (1)
/home/susi/rpmbuild/BUILD/crest-2.11/src/zdata.f90:89:28:

   89 |          procedure print => print_zring !print to screen
      |                            1
Error: ‘::’ needed in PROCEDURE binding with explicit target at (1)
/home/susi/rpmbuild/BUILD/crest-2.11/src/zdata.f90:99:31:

   99 |          procedure allocate => allocate_zgrp
      |                               1
Error: ‘::’ needed in PROCEDURE binding with explicit target at (1)
/home/susi/rpmbuild/BUILD/crest-2.11/src/zdata.f90:100:33:

  100 |          procedure deallocate => deallocate_zgrp
      |                                 1
Error: ‘::’ needed in PROCEDURE binding with explicit target at (1)
/home/susi/rpmbuild/BUILD/crest-2.11/src/zdata.f90:101:29:

  101 |          procedure append => append_zgrp
      |                             1
Error: ‘::’ needed in PROCEDURE binding with explicit target at (1)
/home/susi/rpmbuild/BUILD/crest-2.11/src/zdata.f90:102:28:

  102 |          procedure prgrp => print_zgrp
      |                            1
Error: ‘::’ needed in PROCEDURE binding with explicit target at (1)
/home/susi/rpmbuild/BUILD/crest-2.11/src/zdata.f90:114:31:

  114 |          procedure allocate => allocate_zequal !allocate nat, ord(:) and grp(:)
      |                               1
Error: ‘::’ needed in PROCEDURE binding with explicit target at (1)
/home/susi/rpmbuild/BUILD/crest-2.11/src/zdata.f90:115:29:

  115 |          procedure member => is_x_member       !check if atom x is member of any group
      |                             1
Error: ‘::’ needed in PROCEDURE binding with explicit target at (1)
/home/susi/rpmbuild/BUILD/crest-2.11/src/zdata.f90:116:28:

  116 |          procedure prsum => prsummary_zequal   !print a summary of all the groups
      |                            1
Error: ‘::’ needed in PROCEDURE binding with explicit target at (1)
/home/susi/rpmbuild/BUILD/crest-2.11/src/zdata.f90:117:29:

  117 |          procedure geteng => zequal_geteng     !count number of groups with more than 1 member
      |                             1
Error: ‘::’ needed in PROCEDURE binding with explicit target at (1)
/home/susi/rpmbuild/BUILD/crest-2.11/src/zdata.f90:200:30:

  200 |         procedure allocate =>  allocate_zensemble
      |                              1
Error: ‘::’ needed in PROCEDURE binding with explicit target at (1)
/home/susi/rpmbuild/BUILD/crest-2.11/src/zdata.f90:201:32:

  201 |         procedure deallocate => deallocate_zensemble
      |                                1
Error: ‘::’ needed in PROCEDURE binding with explicit target at (1)
/home/susi/rpmbuild/BUILD/crest-2.11/src/zdata.f90:251:35:

  251 |        call self%zri(i)%deallocate()
      |                                   1
Error: ‘deallocate’ at (1) is not a member of the ‘zring’ structure
/home/susi/rpmbuild/BUILD/crest-2.11/src/zdata.f90:388:30:

  388 |        call self%zri(i)%print(ch)
      |                              1
Error: ‘print’ at (1) is not a member of the ‘zring’ structure
/home/susi/rpmbuild/BUILD/crest-2.11/src/zdata.f90:445:25:

  445 |       call self%allocate(1)
      |                         1
Error: ‘allocate’ at (1) is not a member of the ‘zgrp’ structure
```

**Comment by awvwgk** [comment]
Thanks for reporting. Doesn't look like too much work to fix those issues.

I might have a look once I found time to compile GCC 11 from source.

**Comment by awvwgk** [comment]
Also keep in mind that the GCC fixed were added after version 2.11, you might want to patch the source tar ball with https://github.com/grimme-lab/crest/commit/2674ba27bf7a5f3c20c6f505cdf0f8a852eda1d5 first.

**Comment by susilehtola** [comment]
Thanks for the speedy reply! Build succeeds with the patch.
